Deployment/asdk-prechecker.ps1

```diff
@@ -0,0 +1,384 @@
+            $Global:ChecksFailure++
+        }
+        else
+        {
+            write-host -ForegroundColor green "["(date -format "HH:mm:ss")"]" " -- Check system disk passed successfully."
+            $Global:ChecksSuccess++
+        }   
+}
+
+function CheckDisks {
+    write-host -ForegroundColor yellow "["(date -format "HH:mm:ss")"]" "Checking physical disks..."
+  
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Listing of all physical disks on this server:"
+    write-host -ForegroundColor gray (Get-PhysicalDisk | Format-Table -Property @("FriendlyName", "SerialNumber", "CanPool", "BusType", "OperationalStatus", "HealthStatus", "Usage", "Size") | Out-String)
+    $physicalDisks = Get-PhysicalDisk | Where-Object { ($_.BusType -eq 'RAID' -or $_.BusType -eq 'SAS' -or $_.BusType -eq 'SATA') -and $_.Size -gt 135 * 1024 * 1024 * 1024 }
+    $selectedDisks = $physicalDisks | Group-Object -Property BusType | Sort-Object -Property Count -Descending | Select-Object -First 1
+
+    if ($selectedDisks.Count -ge 3) {
+        write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Listing of all physical disks meeting ASDK requirements:"
+        write-host -ForegroundColor gray ($physicalDisks | Format-Table -Property @("FriendlyName", "SerialNumber", "BusType", "OperationalStatus", "HealthStatus", "Usage", "Size") | Out-String)
+        write-host -ForegroundColor green "["(date -format "HH:mm:ss")"]" " -- Check physical disks passed successfully. Note that ASDK handles situations where there is a pre-existing storage pool, and will delete/recreate it."
+        $Global:ChecksSuccess++
+    }
+
+    if ($selectedDisks.Count -lt 3) {
+        write-host -ForegroundColor red "["(date -format "HH:mm:ss")"]" " -- Check physical disks failed - At least 4 disks or more of the same bus type (RAID/SAS/SATA), and of capacity 135 GB or higher are strongly recommended. 3-disk configurations may work but are not tested by Microsoft."
+        $Global:ChecksFailure++
+     }    
+
+}
+
+function CheckFreeSpaceForExtraction {
+
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " Checking free space for extracting the ASDK files..."
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Listing disks and their free space"
+    write-host -ForegroundColor gray (Get-Disk | Get-Partition | Get-Volume | Sort-Object -Property SizeRemaining -Descending | Out-String)
+    $volumes = (Get-disk | ? {$_.BusType -ne 'File Backed Virtual' -or $_.IsBoot} | Get-Partition | Get-Volume |`
+         ? {-not [String]::IsNullOrEmpty($_.DriveLetter)} | sort -Property SizeRemaining -Descending)
+    if (!$volumes -or ($volumes | Measure-Object).count -le 0) {
+        Write-Host -ForegroundColor red "["(date -format "HH:mm:ss")"]" " -- Free space check failed. No volumes are available."
+        $Global:ChecksFailure++
+    }
+    if ($volumes[0].SizeRemaining -lt 120 * 1024 * 1024 * 1024) {
```

Comment:
Current CloudBuilder.vhdx will be 130GB. Please update to 
`if ($volumes[0].SizeRemaining -lt 130 * 1024 * 1024 * 1024) {`

Reply:
Updated calculation.